vdr/didstore/event.go

```diff
@@ -0,0 +1,137 @@
+/*
+ * Nuts node
+ * Copyright (C) 2022 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package didstore
+
+import (
+	"sort"
+	"time"
+
+	"github.com/nuts-foundation/go-did/did"
+	"github.com/nuts-foundation/nuts-node/crypto/hash"
+	"github.com/nuts-foundation/nuts-node/network/dag"
+)
+
+// event contains the transaction reference and ordering of all DID document diff
+type event struct {
+	// Created is the transaction creation time, used for sorting
+	Created time.Time `json:"created"`
+	// LogicalClock contains the LC header from the transaction
+	LogicalClock uint32 `json:"lc"`
+	// TXRef contains the TX.Ref of the original transaction. Used for ordering
+	TXRef hash.SHA256Hash `json:"txref"`
+	// DocRef contains the reference to the document shelf. Equals transaction payload hash
+	DocRef hash.SHA256Hash `json:"docref"`
+	// MetaRef contains a reference to the documentMetadata shelf. "DID + version"
+	MetaRef string `json:"metaref"`
```

Comment:
confusing, the other `...Ref`s are actual hashes, this isn't

Reply:
The other two follow the way of the rest of the code. This one follows the previous/current version of the metadata keys.